aldor/m4/strict_compile.m4

```diff
@@ -1,12 +1,22 @@
 # Define warnings based on compiler (and version)
 
+AC_DEFUN([ALDOR_CC_OPTION],
+[AC_MSG_CHECKING($CC supports $1); echo > conftest$1.c;
+ res=no
+ if $CC $1 -c conftest$1.c > /dev/null 2>&1; then $2="$1"; res=yes; fi;
+ AC_MSG_RESULT($res)])
+
 AC_DEFUN([ALDOR_STRICT_COMPILE],
-  [AC_MSG_CHECKING(Strict options for C compiler)
-  cfgSTRICTCFLAGS="-pedantic -std=c89 -Wall -Wextra -Werror -Wno-empty-body -Wno-enum-compare -Wno-missing-field-initializers -Wno-sign-compare -Wno-unused -Wno-unused-parameter -Wno-error=format -Wno-error=type-limits -Wno-error=strict-aliasing"
+
+  [ALDOR_CC_OPTION(-Wno-shift-negative-value,cfg_no_shift_negative_value)
```

Comment:
Perhaps we can detect whether the compiler supports this flag even with -Werror enabled. Perhaps the check should always use -Werror to see if the compiler supports the flag.